jupyter-on-gke/variables.tf

```diff
@@ -23,3 +23,47 @@ variable "create_namespace" {
   description = "Enable creation of jupyterhub namespace if it does not exist"
   default     = false
 }
+
+variable "add_auth" {
+  type = bool
+  description = "Enable iap authentication on jupyterhub"
+  default = false
+}
+
+variable "project_id" {
+  type        = string
+  description = "GCP project id"
+  default     = "<>Project ID here"
+}
+
+variable "project_number" {
+  type        = string
+  description = "GCP project number (Not to be confused with porject id)"
+  default     = "<Project Number Here>"
+}
+
+variable "location" {
+  type        = string
+  description = "GCP project location"
+  default     = "us-central1"
+}
+
+variable "client_id" {
```

Comment:
Can we do this programmatically? https://cloud.google.com/iap/docs/programmatic-oauth-clients

Reply:
Eg here: https://github.com/GoogleCloudPlatform/kubeflow-distribution/pull/399/files

But unsure if we need the redirect URI support

Reply:
As discussed offline, will be looking into it separately.